streams/src/test/java/org/apache/kafka/streams/processor/internals/StoreChangelogReaderTest.java

```diff
@@ -371,7 +371,7 @@ public void shouldRestorePartitionsRegisteredPostInitialization() {
     @Test
     public void shouldThrowTaskMigratedExceptionIfEndOffsetGetsExceededDuringRestore() {
         final int messages = 10;
-        setupConsumer(messages, topicPartition);
+        setupConsumer(messages, topicPartition, 6);
```

Comment:
We are cycling forth and back here... This change indicate that our logic is still not correct -- and looking at the code it makes sense:

Here https://github.com/apache/kafka/pull/4300/files#diff-46ed6d177221c8778965ecb1b6657be3R279 we set `nextPosition = record.offset()` what is `5` in this test case and equals to `endOffset` -- but we should return `10` here (and than this test would fail without the change).

We are still mixing up the two cases using a regular changelog topic and the source topic as changelog...